browser/static/ably.js

```diff
@@ -2799,11 +2809,22 @@ var Defaults = {
 	transports:               ['web_socket', 'xhr', 'iframe', 'jsonp']
 };
 
+/* If an environment option is provided, the environment is prefixed to the domain
+   i.e. rest.ably.io with environment sandbox becomes sandbox-rest.ably.io */
+Defaults.environmentHost = function(environment, host) {
+	if (!environment || (String(environment).toLowerCase() === 'production')) {
+		return host;
+	} else {
+		return [String(environment).toLowerCase(), host].join('-');
+	}
+};
+
 Defaults.getHost = function(options, host, ws) {
-	host = host || options.host || Defaults.HOST;
+	var defaultHost = Defaults.environmentHost(options.environment, Defaults.HOST);
+	host = host || options.host || defaultHost;
 	if(ws)
```

Comment:
Unfortunately one line no curly brackets is used throughout so I don't think it's worth changing here.